### DISCUSSION TITLE
Auto expand to current file

### DISCUSSION THREAD

**Discussion by BhasherBEL** [nvim-tree/nvim-tree.lua#2911] [Q&A]
**Is your feature request related to a problem? Please describe.**
When opening a file, i.e. with `nvim a/b/c.sh`, nvim tree shows the current folder, which is really nice, but don't expand to the opened file.

**Describe the solution you'd like**
It would be nice to be able to automatically expend the tree up to the current file, at least at startup, but probably also when using commands such as `:split`, `:edit`, ...

**Describe alternatives you've considered**
For now, I manually expand the tree, but it's not really a long-term solution.

**Comment by alex-courtis** [comment]
You can automatically expand the tree to focus on the current file, with several options to customise this: `:help nvim-tree-opts-update-focused-file`

It's turned off by default.

There are also several recipes here to define startup behaviour: [wiki: Open At Startup](https://github.com/nvim-tree/nvim-tree.lua/wiki/Open-At-Startup)

I reckon "Open For Files And [No Name] Buffers" might suit you.

**Reply by CoinCheung** [reply]
@alex-courtis Would you please show me how to use that function, and how to write it to my init.lua?

**Reply by alex-courtis** [reply]
Before anything [disable netrw Hijacking](https://github.com/nvim-tree/nvim-tree.lua/wiki/Open-At-Startup#netrw-hijacking)
```lua
-- start of your init.lua
vim.g.loaded_netrw       = 1
vim.g.loaded_netrwPlugin = 1
```

Create an [open function](https://github.com/nvim-tree/nvim-tree.lua/wiki/Open-At-Startup#open-for-files-and-no-name-buffers) to match your tastes. This might suit you:
```lua
local function open_nvim_tree(data)

  -- buffer is a real file on the disk
  local real_file = vim.fn.filereadable(data.file) == 1

  if not real_file then
    return
  end

  -- open the tree, find the file but don't focus it
  require("nvim-tree.api").tree.toggle({ focus = false, find_file = true, })
end
```

Create a [hook](https://github.com/nvim-tree/nvim-tree.lua/wiki/Open-At-Startup#recipes) to call that function on vim enter. After everything else:
```lua
local function open_nvim_tree()

  -- open the tree
  require("nvim-tree.api").tree.open()
end
```
